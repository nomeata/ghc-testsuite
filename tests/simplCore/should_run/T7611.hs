import Debug.Trace

myMap :: (a -> a) -> [a] -> [a]
myMap h x = trace "not ok" (map h x)
{-# NOINLINE myMap #-}

{-# RULES "myMap id" myMap (\x -> x) = id #-}

main = do
    putStrLn "myMap id: "
    putStr $ myMap id ""
    putStrLn "myMap (\\x -> x): "
    putStr $ myMap (\x -> x) ""
