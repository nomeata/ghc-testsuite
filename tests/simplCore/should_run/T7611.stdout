myMap id: 
myMap (\x -> x): 
